src/Symfony/Bundle/FrameworkBundle/Tests/Error/PagesDumperTest.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Tests\Error;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Bridge\Twig\ErrorRenderer\TwigErrorRenderer;
+use Symfony\Bundle\FrameworkBundle\Error\PagesDumper;
+use Symfony\Component\ErrorHandler\Exception\FlattenException;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+
+class PagesDumperTest extends TestCase
+{
+    private string $tmpDir = '';
+    private string $tmpCacheDir = '';
+    private ?PagesDumper $pagesDumper = null;
+
+    public function testDumpPagesInGivenDirectory()
+    {
+        $pagesDumper = $this->getPagesDumper();
+        $pagesDumper->dump($this->tmpDir);
+
+        $this->assertFileExists($this->tmpDir.\DIRECTORY_SEPARATOR.'404.html');
+        $this->assertStringContainsString('Error 404', file_get_contents($this->tmpDir.\DIRECTORY_SEPARATOR.'404.html'));
+    }
+
+    public function testDumpPagesInDefaultCacheDirectory()
+    {
+        $pagesDumper = $this->getPagesDumper();
+        $pagesDumper->dump();
+
+        $this->assertFileExists($this->tmpCacheDir.\DIRECTORY_SEPARATOR.'error_pages'.\DIRECTORY_SEPARATOR.'404.html');
+        $this->assertStringContainsString('Error 404', file_get_contents($this->tmpCacheDir.\DIRECTORY_SEPARATOR.'error_pages'.\DIRECTORY_SEPARATOR.'404.html'));
+    }
+
+    protected function getPagesDumper(): PagesDumper
+    {
+        $twigErrorRenderer = $this->createMock(TwigErrorRenderer::class);
+        $twigErrorRenderer
+            ->expects($this->any())
+            ->method('render')
+            ->willReturnCallback(function (...$args) {
+                $this->assertInstanceOf(HttpException::class, $args[0]);
+
+                $exception = FlattenException::createFromThrowable($args[0]);
+                $exception->setAsString(\sprintf('<html><body>Error %s</body></html>', $args[0]->getStatusCode()));
+
+                return $exception;
+            })
+        ;
```

Comment:
What about making assertion using `with($this->isInstanceOf(HttpException::class))` ?
```suggestion
        $twigErrorRenderer
            ->expects($this->once())
            ->method('render')
            ->with($this->isInstanceOf(HttpException::class))
            ->willReturnCallback(function (HttpException $e) {
                $flatten = FlattenException::createFromThrowable($e);
                $flatten->setAsString(\sprintf('<html><body>Error %s</body></html>', $e->getStatusCode()));

                return $flatten;
            })
        ;
```